tests/basilisp/compiler_test.py

```diff
@@ -5425,6 +5426,7 @@ def test_aliased_namespace_not_hidden_by_python_module(
                 """
                 )
             finally:
+                monkeypatch.chdir(cwd)
```

Comment:
Pytest's `monkeypatch` is supposed to  undo this change for us -- are you sure this is necessary?

Reply:
According to the documentation, 

> Use monkeypatch.setenv("PATH", value, prepend=os.pathsep) to modify $PATH, and [monkeypatch.chdir](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.chdir) to change the context of the current working directory during a test.

monkeypatch will change the directory for the duration of the test and then restore it.

but while we are in the test 

``` python
def test_aliased_namespace_not_hidden_by_python_module(
        self, lcompile: CompileFn, monkeypatch: MonkeyPatch
    ):
        with TemporaryDirectory() as tmpdir:
            monkeypatch.chdir(tmpdir)
 # ...
             os.unlink(module_file_path)
```

it will try to delete that directory before it is released by monkeypatch.

Thus the need to restore it before deleting the temp dir.

This is the error that is returned If we don't do this

```
#...
        else:
            if dir_fd is not None:
                raise NotImplementedError("dir_fd unavailable on this platform")
            try:
>               if _rmtree_islink(path):
E               RecursionError: maximum recursion depth exceeded

c:\local\Python311\Lib\shutil.py:752: RecursionError
---------------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------------
WARNING  basilisp.lang.compiler.generator:generator.py:2961 could not resolve a direct link to Var 'require' (basilisp.compiler_test:1)
=============================================================================== short test summary info ================================================================================
FAILED tests/basilisp/compiler_test.py::TestSymbolResolution::test_aliased_namespace_not_hidden_by_python_module - RecursionError: maximum recursion depth exceeded
=================================================================== 1 failed, 572 passed, 2439 deselected in 16.60s ====================================================================
```